eo-runtime/src/test/eo/org/eolang/memory-tests.eo

```diff
@@ -88,3 +95,124 @@
       b.write 20
       a
     $.equal-to 10
+
+[] > memory-is-strictly-typed-int-error
+  memory 42 > m
+  nop > @
+    assert-that
+      try
+        []
+          m.write 3.1416 > @
+        [e]
+          e > @
+        nop
+      $.equal-to "Memory type missmatch: expected <int>, got <float>"
+
+[] > memory-is-strictly-typed-int-error-overflow
+  memory 256 > m
+  nop > @
+    assert-that
+      try
+        []
+          m.write (9223372036854775807.plus 1) > @
+        [e]
+          e > @
+        nop
+      $.equal-to "Not enough memory to write: expected <2^63 - 1>, got <2^63>"
```

Comment:
@Graur Should we throw an exception in that case? Maybe it's better just to write `-9,223,372,036,854,775,808`?

Reply:
@volodya-lombrozo I like ti, thanks!